aggregator_core/src/datastore/tests.rs

```diff
@@ -126,7 +126,7 @@ async fn retry_limit() {
                 let num_runs = Arc::clone(&num_runs);
                 Box::pin(async move {
                     num_runs.fetch_add(1, Ordering::Relaxed);
-                    tx.retry();
+                    tx.retry.store(true, Ordering::Relaxed); // artificially force a retry
```

Comment:
More ammunition for reverting the retry change. If in test we have to induce something that can't actually be done in user code, since tx.retry is private, we might as well ditch that code.

Reply:
We can still retry based on errors received from Postgres; think of this as a very cheap way to emulate fault-injection into our connection to Postgres.